src/hello_client.erl

```diff
@@ -165,12 +166,17 @@ handle_call(terminate, _From, State) ->
 %% @hidden
 handle_info({?INCOMING_MSG, Message}, State) ->
     incoming_message(Message, State);
+handle_info(?PING, State = #client_state{ping_counter = PingCounter}) when PingCounter >= 3 ->
+    ?LOG_ERROR("There is no PONG answer on PING for a long time. Client will be restarted.", []),
```

Comment:
"for a long time" - seriously? can we please be more specific?
it seems you are just pretending time-measurement but instead you are counting missed ping/pong.
how about being precise and honest in one go? keep track of the timestamp of the last successful ping/pong and compare that to now(). you will have a precise time-period to expose for the log-reader and the tolerance can be configured directly as a time-span (instead of some combination of PingCounter and PingInterval).